src/libcore/mem/mod.rs

```diff
@@ -88,16 +121,17 @@ pub use crate::intrinsics::transmute;
 /// // `s` is implicitly dropped and its memory deallocated.
 /// ```
 ///
-/// Using `ManuallyDrop` here has two advantages:
+/// `ManuallyDrop` robustly prevents double-free because we disable `v`'s destructor
+/// before doing anything else. `mem::forget()` doesn't allow this because it consumes its
+/// argument, forcing us to call it only after extracting anything we need from `v`.  Even
```

Comment:
Sometimes you have one space before `.` and sometimes two. Please be consistent. Generally, the style is to use 1 space.

Reply:
No prob, I'll update it tonight.